daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala

```diff
@@ -875,6 +875,114 @@ class DecodeV1Spec
     }
   }
 
+  "decodeChoice" should {
+    val stringTable = ImmArraySeq("SomeChoice", "controllers", "observers", "self", "arg", "body")
+    val typeTable = ImmArraySeq(TUnit)
+    val templateName = Ref.DottedName.assertFromString("Template")
+    val controllersExpr = DamlLf1.Expr.newBuilder().setVarInternedStr(1).build()
+    val observersExpr = DamlLf1.Expr.newBuilder().setVarInternedStr(2).build()
+    val bodyExp = DamlLf1.Expr.newBuilder().setVarInternedStr(5).build()
+
+    "reject choice with observers if lf version < 1.7" in {
```

Comment:
Shouldn't this be
```suggestion
    "reject choice with observers if lf version < 1.dev" in {
```

Reply:
No.  I have to build an special case for 1.6, where we do not have string interning. 
:cry:

Reply:
Oh no, I'm sorry. Do we have other cases like this where we could extract a utility function?

Reply:
There currently no other tests for `TemplateChoice` so we cannot factorize.